Tasks/VSMobileCenterTest/vsmobilecentertest.ts

```diff
@@ -0,0 +1,213 @@
+        }
+        else if (framework === 'calabash') {
+            prepareRunner.arg(['--app-path', app]);
+            addStringArg('--project-dir', 'calabashProjectDir', true, prepareRunner);
+            addStringArg('--sign-info', 'signInfo', false, prepareRunner);
+            addStringArg('--config', 'calabashConfigFile', false, prepareRunner);
+            addStringArg('--profile', 'calabashProfile', false, prepareRunner);
+            addBooleanArg('--skip-config-check', 'calabashSkipConfigCheck', prepareRunner);
+        } 
+        else if (framework === 'uitest') {
+            prepareRunner.arg(['--app-path', app]);
+            addStringArg('--build-dir', 'uitestBuildDir', true, prepareRunner); 
+            addStringArg('--store-file', 'uitestStoreFile',false, prepareRunner);
+            addStringArg('--store-password', 'uitestStorePass',false, prepareRunner);
+            addStringArg('--key-alias', 'uitestKeyAlias',false, prepareRunner);
+            addStringArg('--key-password', 'uitestKeyPass',false, prepareRunner);
+            addStringArg('--uitest-tools-dir', 'uitestToolsDir',false, prepareRunner);
+            addStringArg('--sign-info', 'signInfo', false, prepareRunner); 
+        }
+
+        // append user defined inputs
+        let prepareOpts = tl.getInput('prepareOpts', false);
+        if (prepareOpts) {
+            prepareRunner.line(prepareOpts);
+        }
+
+        if (debug) {
+            prepareRunner.arg('--debug');
+        }
+        prepareRunner.arg('--quiet');
+
+        // Get build inputs
+        let testRunner = tl.tool(cliPath);
+        testRunner.arg(['test', 'run', 'manifest']);
+        testRunner.arg(['--manifest-path', `${path.join(artifactsDir, 'manifest.json')}`]);
+        testRunner.arg(['--app-path', app, '--app', appSlug]);
+
+        addStringArg('--devices', 'devices', true, testRunner); 
+        addStringArg('--test-series', 'series', false, testRunner); 
+        addStringArg('--dsym-dir', 'dsymDir', false, testRunner); 
+        addBooleanArg('--async', 'async', testRunner); 
+
+        let locale: string = tl.getInput('locale', true);
+        if (locale === 'user') {
+            tl.debug('Use user defined locale.'); 
+            locale = tl.getInput('userDefinedLocale', true);
+        } 
+        testRunner.arg(['--locale', locale]);
+
+        let options: string = tl.getInput('opts', false);
+        if (options) {
+            testRunner.line(options);
+        }
+        if (debug) {
+            testRunner.arg('--debug');
+        }
+        testRunner.arg('--quiet');
+        if (credsType === 'serviceEndpoint'){
+            // add api key
+            let apiToken = getEndpointAPIToken('serverEndpoint');
+            testRunner.arg(['--token', apiToken]);
+        }
+
+        // login if necessary
+        if (credsType === 'inputs') {
+            await loginRunner.exec();
+            loggedIn = true;
+        }
+
+        // test prepare
```

Comment:
I agree in general.   The reason I am trying to setup first here is we can do some basic input validation first just to make sure user doesn't need to wait a long time before an obvious error is encountered.